go-controller/hybrid-overlay/pkg/controller/node.go

```diff
@@ -57,6 +59,9 @@ func nodeChanged(old, new interface{}) bool {
 
 	oldCidr, oldNodeIP, oldDrMAC, _ := getNodeDetails(oldNode)
 	newCidr, newNodeIP, newDrMAC, _ := getNodeDetails(newNode)
+	if newNode.Labels[v1.LabelOSStable] == string(v1.Linux) {
```

Comment:
@JacobTanenbaum do we really need this Linux check? I think we can do this for Windows nodes too; if the key doesn't exist in the map (which it wont for new or old) then reflect.DeepEqual() will return true:

```
	a := map[string]string{}
	b := map[string]string{}
	f := reflect.DeepEqual(a["foobar"], b["foobar"])
	fmt.Printf("f %v\n", f)
```

prints "f true" for example. I'd just add the annotation check to the end of the existing one and call it a day.

Reply:
done